### ISSUE TITLE
RLVR pg_loss goes up over training

### ISSUE THREAD

**Issue by terarachang** [allenai/open-instruct#836]
Hi,
Thanks for the nice repo! I'm trying to reproduce the results of `allenai/Llama-3.1-Tulu-3-8B` by running `bash scripts/train/rlvr/tulu_rlvr.sh` on a node of 8 GPUs. I shortened the `total_episodes` while keeping all the other hyperparameters the same as the script.
I noticed that the verifiable correct rate fluctuates a lot during training and the `loss/policy_avg` goes up over steps, which seems problematic.

<img width="717" height="330" alt="Image" src="https://github.com/user-attachments/assets/e267db5a-d495-42f1-96c3-6e38e5358f2b" />
<img width="723" height="336" alt="Image" src="https://github.com/user-attachments/assets/6e7f6fff-99a1-46e1-864f-a4c0a0d7e1ca" />

Correct me if I'm wrong, my understanding is that `loss/policy_avg` tracks the [pg_loss](https://github.com/allenai/open-instruct/blob/main/open_instruct/ppo_vllm_thread_ray_gtrl.py#L1374), which is `-advantage * ratio` before PPO clipping. Therefore, we should expect this loss to go more and more negative in healthy training.

Thank you!

**Comment by hamishivi** [comment]
Actually, I believe I often see the loss is somewhat flat, or generally fluctuates a bit (sometimes even initially goes up) in most of my runs (and in runs in other frameworks like verl)! Since we are sampling (roughly) online, the objective we are improving is shifting as we train (since it is reliant on the advantages of our online samples).

As for the initial increase, I believe this happens because the average advantage seems to grow a little initially?

**Comment by terarachang** [comment]
Thanks! Yeah I understand that loss terms in RL usually don't mean much. But my reward is also not increasing...
